### PR TITLE
fix: Space Activity Stream Load more issue - EXO-66818 - Meeds-io/meeds#1199

### PR DESCRIPTION
**Pull request by sofyenne**
Before this change, when attempting to load more than 50 space-posted activities, an  HeadersTooLargeException exception would be raised. This issue was caused by the addPreloadActivityIds method, which added the entire list of Activity IDs to the response headers when the preload limit value exceeded the length of the activity IDs list. This change resolves this issue by recalculating the limit and offset in the specific case where the limit is greater than the length of the activity IDs list

### CODE REVIEW

File: component/service/src/main/java/org/exoplatform/social/rest/impl/activity/ActivityRestResourcesV1.java [hunk 1]
```diff
@@ -1076,7 +1076,7 @@ private List<DataEntity> convertToEntities(List<ExoSocialActivity> activities,
   private void addPreloadActivityIds(List<String> activityIds, int preloadLimit, String expand, ResponseBuilder responseBuilder) {
     int offset = preloadLimit > MAX_TO_PRELOAD ? preloadLimit - MAX_TO_PRELOAD : 0;
     List<String> preloadActivityIds =
-                                    activityIds.size() >= preloadLimit ? activityIds.subList(offset, preloadLimit) : activityIds;
+                                    activityIds.size() >= preloadLimit ? activityIds.subList(offset, preloadLimit) :  activityIds.subList(offset > MAX_TO_PRELOAD ? offset : 0, activityIds.size());
```
Comment (ahamdi) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/activity/ActivityRestResourcesV1.java]: recalculate the preload limit the salme as the offset in a dedicated statement to make the code more readable

**Review by ahamdi — CHANGES_REQUESTED** [review]
Could you add a Unit test ?

File: component/service/src/main/java/org/exoplatform/social/rest/impl/activity/ActivityRestResourcesV1.java (hunk 1 above)
Comment (sofyenne) [reply component/service/src/main/java/org/exoplatform/social/rest/impl/activity/ActivityRestResourcesV1.java]: done !

**Comment by exo-swf** [comment]
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

**Review by ahamdi — APPROVED** [review]

**Pull request merged**